### PR TITLE
[Java] Prepend ":job_id:<jobid>" to java-worker-<jobid>-<pid>.log to make Java logging consistent with Python

### LINKED ISSUE #33625: [Java] Prepend ":job_id:<jobid>" to java-worker-<jobid>-<pid>.log to make Java logging consistent with Python
### Description

In ray 2.3, there is PR [#31772](https://github.com/ray-project/ray/pull/31772)  to get real job id from log file content instead of log file name. You can check log_monitor.py. If there is no job id (pattern: ":job_id:") found from log file content, then job_id field of log message object is None. For log message without job_id set, ray just blindly prints it out in console.

To make Java logging consistent with the PR [#31772](https://github.com/ray-project/ray/pull/31772), we can add some logic to the end of "setupLogging" method in LoggingUtil, to write a line with content, ":job_id: real job id\n", to log file.

### Use case

To make Java logging consistent with PR [#31772](https://github.com/ray-project/ray/pull/31772) which seems for lazy worker binding. Otherwise, we may print too many logs from different drivers in shell console, just like issue we experienced in raydp with ray 2.3. Too many spark executors logs get printed in console.

### PR DESCRIPTION
**Pull request by jiafuzha**
## Why are these changes needed?

To make Java logging consistent with PR https://github.com/ray-project/ray/pull/31772 which seems for lazy worker binding. Otherwise, we may print too many logs from different drivers in shell console.

## Related issue number

Closes #33625

## Checks

- [*] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [*] I've run `scripts/format.sh` to lint the changes in this PR.
- [*] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [*] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zjf2012** [comment]
@kira-lin please help review.

**Comment by kira-lin** [comment]
@jovany-wang please help review

File: java/runtime/src/main/java/io/ray/runtime/util/LoggingUtil.java [hunk 1]
```diff
@@ -78,18 +79,20 @@ public static synchronized void setupLogging(RayConfig rayConfig) {
...
-              javaWorkerLogName,
-              "java-worker-" + jobIdHex + "-" + SystemUtil.pid(),
-              config.getString("ray.logging.pattern")),
+              javaWorkerLogName, logFileName, config.getString("ray.logging.pattern")),
           maxFileSize,
           maxBackupFiles,
           null);
       rootLoggerBuilder.add(globalConfigBuilder.newAppenderRef(javaWorkerLogName));
       globalConfigBuilder.add(rootLoggerBuilder);
+      // write :job_id:<job_id> to the beginning of log file to conform
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/util/LoggingUtil.java:93]:
```suggestion
      // write `:job_id:<job_id>` to the beginning of log file to conform
```

**Review by jovany-wang — APPROVED** [review]
LGTM

**Comment by jovany-wang** [comment]
Thanks for the contribution. I'll merge it once CI passes.

**Comment by zjf2012** [comment]
@jovany-wang Most of test succeeded. Some of them failed, but not related to my PR. Do I need to merge master and try again? Or you can merge it for now?

**Comment by jovany-wang** [comment]
Yes. The failures are unrelated. I'm merging it.

**Comment by zjf2012** [comment]
> Yes. The failures are unrelated. I'm merging it.

thanks

**Pull request merged**